### PR TITLE
Fix unrecognized selector error

### PR DESCRIPTION
**Pull request by giladno**
Fixes "[NSTaggedPointerString unsignedIntValue]: unrecognized selector" error

### CODE REVIEW

**Comment by chirag04** [comment]
cc @gre @janicduplessis

File: React/Base/RCTMultipartStreamReader.m [hunk 1]
```diff
@@ -76,7 +76,7 @@ - (void)emitProgress:(NSDictionary *)headers
   // Throttle progress events so we don't send more that around 60 per second.
   CFTimeInterval currentTime = CACurrentMediaTime();
 
-  NSUInteger headersContentLength = headers[@"Content-Length"] != nil ? [headers[@"Content-Length"] unsignedIntValue] : 0;
```
Comment (chirag04) [inline React/Base/RCTMultipartStreamReader.m]: `unsignedIntegerValue` maybe?

**Comment by gre** [comment]
It works! I don't think you need the extra cast tho. just `[headers[@"Content-Length"] intValue]` works ;)

**Comment by gre** [comment]
`unsignedIntegerValue` seems to break too. weird that XCode seems to say it's fine but it breaks at runtime.

**Comment by giladno** [comment]
@gre the extra casting is only to avoid compilation warning by xcode

**Comment by giladno** [comment]
@gre unsignedIntegerValue is a selector of NSNumber, not NSString

**Comment by gre** [comment]
@giladno , mmh look again the code:

```
(NSUInteger)[headers[@"Content-Length"] intValue]
```

you cast the result of the whole thing, not of the header.

intValue is on NSString too

**Comment by gre** [comment]
I don't think the cast is helping anything because it's a `NSUInteger headersContentLength` variable anyway ?

File: React/Base/RCTMultipartStreamReader.m (hunk 1 above)
Comment (gre) [reply React/Base/RCTMultipartStreamReader.m]: unfortunately no, both `unsignedIntValue` and `unsignedIntegerValue` breaks (at runtime)

**Comment by giladno** [comment]
@gre you assign an int to unsigned int - xcode will complain unless you cast. This is only to avoid the warning, it won't break the compilation. The casting is optional...

**Comment by gre** [comment]
oops sorry, I didn't follow the conversation clearly :) ok for the warns ;)

**Comment by usakc** [comment]
Hi,
here is quick fix for this error:
`"postinstall": "sed -i '' 's/\\[headers\\[@\"Content-Length\"\\] unsignedIntValue\\]/\\[headers\\[@\"Content-Length\"\\] intValue\\]/' ./node_modules/react-native/React/Base/RCTMultipartStreamReader.m"`
Add it to "scripts" in package.json in your project.

Hope this help someone :).

File: React/Base/RCTMultipartStreamReader.m (hunk 1 above)
Comment (usakc) [reply React/Base/RCTMultipartStreamReader.m]: I think that the statement can be changed to
NSInteger headersContentLength = headers[@"Content-Length"] != nil ? [headers[@"Content-Length"] intValue] : 0; so the value didn't have to be cast.

**Comment by janicduplessis** [comment]
I'm confused about how this worked for me... This being should be just fine.

**Comment by ide** [comment]
For some reason the value in the dictionary is an NSString, and NSStrings don't define unsignedIntValue.

BTW this should be either `NSInteger = value.integerValue` or `int = value.intValue` to make the types line up.

**Comment by shergin** [comment]
@ide Yeah, you are right. PR is welcome. :)

**Pull request closed without merging**